vm/src/object/core.rs

```diff
@@ -115,7 +115,7 @@ struct PyInner<T> {
     weak_list: WeakRefList,
 
     payload: T,
-    slots: Vec<Option<PyObjectRef>>,
+    slots: PyRwLock<Vec<Option<PyObjectRef>>>,
```

Comment:
Are slots editable after initialization? Maybe we only need `Box<[Option<PyObjectRef>]>`. I am not sure this is related to the problem.

Reply:
Sorry, the above comment is incorrect. I made the `slots`' type as `Box<[PyRwLock<Option<PyObjectRef>>]>` and I saw the situation the slots' length is too big though it was correct when creating the slice. And I tried `Vec<T>` also but it also does. I researched more and I guess it's from 

In `_HashedSeq` case, it has `slots: vec{len: 1, cap: 1}`. But after `Into::into` (`From<PyRef<T>> for PyObjectRef`) it became to have `slots: vec{len: 0, cap: 8}` (random). And in the `PyObject.set_slot`, it becomes to have `slots: vec{len: 4, cap: <SOME_BIG_INT e.g. 4396189216>}`. 😭

Reply:
When I tried to see the memory view with vscode codelldb extension, that's same with the screenshot.
<img width="1350" alt="image" src="https://user-images.githubusercontent.com/26626194/190836915-9d768463-9094-4587-8a9f-2a093ee6908b.png">

Reply:
It was a problem related to struct fields order (`repr(C)`). I resolved in [f140794](https://github.com/RustPython/RustPython/pull/4160/commits/f1407944c6edd29c33141ba5485e446c674a6e21)